cmd/sonobuoy/app/images.go

```diff
@@ -165,124 +176,180 @@ func getClusterVersion(kubeconfig Kubeconfig) (string, error) {
+				Dst: substituteRegistry(config.DefaultSystemdLogsImage, customRegistry),
+			})
+		case e2ePlugin:
+			version, err := getClusterVersion(kubeconfig)
+			if err != nil {
+				return []error{errors.Wrap(err, "failed to get cluster version")}
+			}
 
-		defaultImages, err := image.GetE2EImages(defaultE2ERegistries, version)
-		if err != nil {
-			return []error{errors.Wrap(err, "couldn't get images")}
-		}
+			tagPairs, err := image.GetE2EImageTagPairs(e2eRegistryConfig, version)
+			if err != nil {
+				return []error{errors.Wrap(err, "couldn't...something")}
+			}
 
-		privateImages, err := image.GetE2EImages(e2eRegistryConfig, version)
-		if err != nil {
-			return []error{errors.Wrap(err, "couldn't get images")}
+			conformanceImage := resolveConformanceImage(version)
+			imagePairs = append(imagePairs, image.TagPair{
+				Src: conformanceImage,
+				Dst: substituteRegistry(conformanceImage, customRegistry),
+			})
+			imagePairs = append(imagePairs, tagPairs...)
+		default:
+			return []error{errors.Errorf("Unsupported plugin: %v", plugin)}
 		}
+	}
 
-		imageClient := image.NewImageClient()
+	imageClient := image.NewImageClient()
+	return imageClient.PushImages(imagePairs, numDockerRetries)
+}
 
-		return imageClient.PushImages(defaultImages, privateImages, numDockerRetries)
-	default:
-		return []error{errors.Errorf("Unsupported plugin: %v", plugin)}
+func deleteImages(plugins []string, kubeconfig Kubeconfig, e2eRegistryConfig string) []error {
+	images := []string{
+		config.DefaultImage,
 	}
-}
+	for _, plugin := range plugins {
+		switch plugin {
+		case systemdLogsPlugin:
+			images = append(images, config.DefaultSystemdLogsImage)
+		case e2ePlugin:
+			version, err := getClusterVersion(kubeconfig)
+			if err != nil {
+				return []error{errors.Wrap(err, "failed to get cluster version")}
+			}
 
-func deleteImages(plugin string, kubeconfig Kubeconfig, e2eRegistryConfig string) []error {
-	switch plugin {
-	case "e2e":
-		version, err := getClusterVersion(kubeconfig)
-		if err != nil {
-			return []error{errors.Wrap(err, "failed to get cluster version")}
-		}
+			e2eImages, err := image.GetE2EImages(e2eRegistryConfig, version)
+			if err != nil {
+				return []error{errors.Wrap(err, "couldn't get images")}
+			}
 
-		images, err := image.GetE2EImages(e2eRegistryConfig, version)
-		if err != nil {
-			return []error{errors.Wrap(err, "couldn't get images")}
+			images = append(images, resolveConformanceImage(version))
+			images = append(images, e2eImages...)
+		default:
+			return []error{errors.Errorf("Unsupported plugin: %v", plugin)}
 		}
+	}
 
-		imageClient := image.NewImageClient()
+	imageClient := image.NewImageClient()
+	return imageClient.DeleteImages(images, numDockerRetries)
+}
 
-		return imageClient.DeleteImages(images, numDockerRetries)
+func substituteRegistry(image string, customRegistry string) string {
+	trimmedRegistry := strings.TrimRight(customRegistry, "/")
+	components := strings.SplitAfter(image, "/")
```

Comment:
I thought I left a comment on this but it's gotten lost :confused: 

This is a best attempt at replacing the registry in a full image name. It's possible for the "name" component of an image name to also contain a `/` and we see that in [some of the E2E test images](https://github.com/vmware-tanzu/sonobuoy/blob/master/pkg/image/v1.17.go#L71-L74). In cases like these, this function would substitute everything up to the last `/` which could include the registry *and* part of the image name. I don't think there's a way to identify what is the registry URL component of a FQIN. The images will still tag and push correctly, but it might be confusing for users when replacing images in their plugin definitions.

Reply:
Its a short function but I think this is a good function to have in a function so that we can visually ensure it handles certain cases.